src/test/java/com/skynest/EditUserByIdTest.java

```diff
@@ -30,7 +38,40 @@ void valid_edit_user_by_id_test(EditRequest editRequest) throws IOException {
 
     @DataProvider(name = "UserData")
     public Object[][] getUserData() {
-        return new Object[][]{
-                new Object[]{EditRequest.generateValidEditRequest()}};
+        return new Object[][]{new Object[]{EditRequest.generateValidEditRequest()}};
+    }
+
+    @Test(dataProvider = "UserData")
+    void edit_random_worker_details_as_admin_test(EditRequest editRequest) throws IOException {
+        loginAs(Roles.ADMIN);
+        Response getAllUsersResponse = skyNestBackendClient.getAllUsers();
+        getAllUsersResponse.then().statusCode(SC_OK);
+
+        List<UserResponse> userResponses = JsonTransformer.mapResponseToList(getAllUsersResponse, UserResponse.class);
+        UUID firstWorkerId = getRandomWorkerId(userResponses);
+
+
+        Response editUserResponse = skyNestBackendClient.editUserById(editRequest, firstWorkerId);
+        editUserResponse.then().statusCode(SC_OK);
+
+        EditResponse editResponse = editUserResponse.as(EditResponse.class);
+
+        Assert.assertNotNull(editResponse.getId());
+
+        Assert.assertEquals(editResponse.getName(), editRequest.getName());
+        Assert.assertEquals(editResponse.getSurname(), editRequest.getSurname());
+        Assert.assertEquals(editResponse.getPhoneNumber(), editRequest.getPhoneNumber());
+        Assert.assertEquals(editResponse.getAddress(), editRequest.getAddress());
+        Assert.assertEquals(editResponse.getPositionInCompany(), editRequest.getPositionInCompany());
+    }
+
+    private UUID getRandomWorkerId(List<UserResponse> userResponses) {
+        for (int i = 0; i < userResponses.size(); i++) {
```

Comment:
more Java8+ style maybe?
ping @UkiWoDao 

```   private UserResponse getRandomWorkerId(List<UserResponse> userResponses) {
        Optional<UserResponse> first = userResponses.stream().filter(userResponse -> 
            userResponse.getEmail().equals(WORKER_EMAIL)).findAny();
        
        return first.orElse(null);
    }```

Reply:
@kantarogit I have very limited experience with optionals and the stream api in general. I'm just going to defer to you on this one and go ahead and educate myself on the matter :)

Reply:
Well, I am not familiar with Java8+ style, but will ping Uros for help if needed :D

Reply:
i think this snipped would be sufficient for now (take a note to go into details into stream api, lambdas and optionals later) :)

not a priority atm, but would really recommend using stream api for collections as it make the whole process less prone to errors